src/com/hms/flexyosisoftconnector/PayloadBuilder.java

```diff
@@ -298,7 +371,20 @@ public static String getTypeBody() {
    * @return returns the JSON segment to construct a container for each of the tags.
    */
   public static String getContainerSettingJson(int startTagIndex, int numToProccess) {
-    String typeID = "HMS-type-" + getFlexyName();
+    String typeID;
+    if (((TagInfo) TagInfoManager.getTagInfoList().get(i)).getType() == TagType.FLOAT) {
```

Comment:
((TagInfo) TagInfoManager.getTagInfoList().get(i)).getType() Should be stored in a variable so you dont make repeat calls to it